src/ru/itmo/kotlin/plugin/fir/checkers/InjectedConstructorChecker.kt

```diff
@@ -0,0 +1,23 @@
+package ru.itmo.kotlin.plugin.fir.checkers
+
+import org.jetbrains.kotlin.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.FirSession
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.declaration.FirClassChecker
+import org.jetbrains.kotlin.fir.declarations.FirClass
+import org.jetbrains.kotlin.fir.declarations.hasAnnotation
+import org.jetbrains.kotlin.fir.declarations.primaryConstructorIfAny
+import org.jetbrains.kotlin.name.ClassId
+import org.jetbrains.kotlin.name.Name
+import ru.itmo.kotlin.plugin.fir.DependencyInjector
+
+class InjectedConstructorChecker(private val session: FirSession) : FirClassChecker() {
```

Comment:
There is no need to make checker a `class` instead of `object`
`sessions` is passed inside `CheckerContext`